CBinding/CLang/CLangManager.cs

```diff
@@ -0,0 +1,482 @@
+﻿using System;
+using CBinding;
+using ClangSharp;
+using MonoDevelop.Ide;
+using MonoDevelop.Projects;
+using System.Collections.Generic;
+using System.Threading;
+using MonoDevelop.Ide.Editor;
+using System.Runtime.InteropServices;
+using CBinding.Refactoring;
+using CBinding.Parser;
+using System.IO;
+
+namespace CBinding
+{
+	/// <summary>
+	/// Class to manage clang Translation units, thread safe
+	/// </summary>
+	public class CLangManager : IDisposable
+	{
+		public readonly object syncroot = new object ();
+		private CProject project;
```

Comment:
Please leave off the redundant `private` accessibility modifier as per the MonoDevelop code guidelines: http://www.monodevelop.com/developers/#coding-guidelines

If you have source analysis enabled, it should hint about these (not 100% sure this rule was ported to Roslyn though).